plat/hisilicon/hikey/hikey_ddr.c

```diff
@@ -923,38 +1409,39 @@ static void init_ddrc_qos(void)
 	mmio_write_32((0xf7124000 + 0x0d0), 0x3020100);
 }
 
-static void ddr_phy_reset(void)
-{
-	mmio_write_32(0xf7030340, 0xa000);
-	mmio_write_32(0xf7030344, 0xa000);
-}
-
 void hikey_ddr_init(void)
 {
 	uint32_t data;
+	uint32_t ddr_freq;
 
 	init_pll();
 	init_freq();
 
-	/*
-	 * Init DDR with 533MHz. Otherwise, DDR initialization
-	 * may fail on 800MHz on some boards.
-	 */
-	ddr_phy_reset();
-	init_ddr(DDR_FREQ_533M);
-	/* Init DDR with 800MHz. */
-	ddr_phy_reset();
-	init_ddr(DDR_FREQ_800M);
-
+	ddr_freq = DDR_FREQ_800M;
```

Comment:
This function is a bit weird. Maybe you could make `ddr_freq` a parameter of the function. It won't change much, though, there will still be unused code.

Reply:
I'll update it.